Homework5/Controllers/PeopleApiController.cs

```diff
@@ -0,0 +1,198 @@
+    {
+        private static List<Person> personList = new List<Person>();
+
+        [HttpPost]
+        public IActionResult AddPerson(Person personObj)
+        {
+            personList.Add(personObj);
+
+            return Created("/peopleapi", personObj);
+        }
+
+        [HttpGet]
+        public IActionResult GetAllPersons()
+        {
+            return Ok(personList);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetPersonById(int id)
+        {
+            Person tempPersonObj = FindPersonById(id);
+
+            if (tempPersonObj == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+
+            return Ok(tempPersonObj);
+        }
+
+        [HttpPut("{id}")]
+        public IActionResult AddPetToPersonChosenById(int id, Pet petObj)
+        {
+            Person tempPersonObj = FindPersonById(id);
+            if (tempPersonObj == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+
+            tempPersonObj.Pets.Add(petObj);
+
+            return Ok(tempPersonObj);
+        }
+
+        [HttpDelete("{id}")]
+        public IActionResult RemovePerson(int id)
+        {
+            Person tempPersonObj= FindPersonById(id);
+
+            if (tempPersonObj == null)
+            {
+                return NotFound($"Shopping list by id {id} was not found.");
+            }
+
+            personList.Remove(tempPersonObj);
+
+            return Ok();
+        }
+
+        [HttpPut("{id}/{petName}")]
+        public IActionResult RemovePetFromPerson(int id, string petName)
+        {
+            Person tempPersonObj = FindPersonById(id);
+            if (tempPersonObj == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+
+            Pet tempPetObj = FindPetByName(tempPersonObj.Pets, petName);
+            if (tempPetObj == null)
+            {
+                return NotFound($"Pet by name of {petName} was not found.");
+            }
+
+            RemovePetObjectFromPersonObject(tempPersonObj, tempPetObj);
+            tempPersonObj = FindPersonById(id);
+            return Ok(tempPersonObj);
+        }
+
+        [HttpPatch("{id}/updatePerson")]
+        public IActionResult UpdatePersonDetails(int id, Person personObj)
+        {
+            if (FindPersonById(id) == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+
+            UpdatePersonObjectInList(id, personObj);
+            return Ok();
+        }
+
+        [HttpPut("{id}/{spouseId}/marry")]
+        public IActionResult MarryPersons(int id,int spouseId) 
+        {
+            if (FindPersonById(id) == null || FindPersonById(spouseId) == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+            ChangeSpouseIds(id, spouseId);
+
```

Comment:
I don't understand why this needs to be moved to Person class, as this function changes two different objects so it still needs to query through available object list to get correct spouses. Also it would require a list of objects be passed to when in person class. Unless I'm missing something obvious

Reply:
You're right. Another person had Marry function or similar, which changed ids of already found person and existing person to update them accordingly. I think you could spli this function into finding the person and then updating the ids on both through that method. Updating two ids is very simple and does not reduce complexity, but method called `Mary` clarifies the intention of two id changes for spouses.